osu.Game/Overlays/Comments/CommentEditor.cs

```diff
@@ -167,33 +170,59 @@ private void load(OverlayColourProvider colourProvider)
         protected override void LoadComplete()
         {
             base.LoadComplete();
-            Current.BindValueChanged(_ => updateCommitButtonState(), true);
-            apiState.BindValueChanged(updateStateForLoggedIn, true);
+            Current.BindValueChanged(_ => updateState());
+            apiState.BindValueChanged(_ => Scheduler.AddOnce(updateState));
+            CommentableMeta.BindValueChanged(_ => Scheduler.AddOnce(updateState));
+            updateState();
         }
 
         protected abstract void OnCommit(string text);
 
-        private void updateCommitButtonState() =>
-            commitButton.Enabled.Value = loadingSpinner.State.Value == Visibility.Hidden && !string.IsNullOrEmpty(Current.Value);
-
-        private void updateStateForLoggedIn(ValueChangedEvent<APIState> state) => Schedule(() =>
+        private void updateState()
         {
-            bool isAvailable = state.NewValue > APIState.Offline;
+            bool isOnline = apiState.Value > APIState.Offline;
+            var canNewCommentReason = CommentEditor.canNewCommentReason(CommentableMeta.Value);
```

Comment:
Probably use explicit type here.

Reply:
You can't, it'd raise an inspection :sweat_smile: The reason the fully qualified name is here is because the local variable and the static method share the same identifier. I guess I could rename the local if it works for you?

Reply:
Ohhh I completely misread that. Sure.